provider/serialization/casts.go

```diff
@@ -0,0 +1,179 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright 2024 FeatureForm Inc.
+//
+
+package serialization
+
+import (
+	"fmt"
+	"strconv"
+)
+
+func CastNumberToFloat32(value any) (float32, error) {
+	// I have to do one type per case for this to work properly.
+	switch typed := value.(type) {
+	case int:
+		return float32(typed), nil
+	case int32:
+		return float32(typed), nil
+	case int64:
+		return float32(typed), nil
+	case int8:
+		return float32(typed), nil
+	case int16:
+		return float32(typed), nil
+	case float32:
+		return typed, nil
+	case float64:
```

Comment:
q: do we care at all about precision loss?

Reply:
I think there's 2 parts to this answer:
1. This should probably work out, as ValueType during serialization should ensure that whatever is being serialized wouldn't lose precision for any of the types, or throw an error potentially.
2. This turned out to be completely unused in firestore, but did happen to clean up the DDB provider (by refactoring it out to a separate package), so it was kept in the PR anyway.
We should probably revisit this, but since Firestore doesn't touch it I feel like this is outside of the scope of the PR.

Reply:
gotcha resolve at will, although maybe fiddle with a throw away unit test to prove out #1.